### PR TITLE
Upgraded web-forms to 0.6

### PR DESCRIPTION
**Pull request by sadiqkhoja**
#### What has been done to verify that this works as intended?

Manual verification, deployed at Dev environment. Sample Form: https://dev.getodk.cloud/#/projects/2/forms/ExternalInstance

#### Why is this the best possible solution? Were any other approaches considered?

Using the existing `request` object to handle attachment fetching seemed like a logical option to me. The other options were to create a new function from scratch using `fetch` or import `XFormResource` from `@getodk/common`, both options would have required separate authentication and error handling mechanism.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

None

#### Does this change require updates to user documentation? If so, please file an issue [here](https://github.com/getodk/docs/issues/new) and include the link below.

No

#### Before submitting this PR, please make sure you have:

- [x] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: src/components/form/preview.vue [hunk 1]
```diff
@@ -84,6 +86,17 @@ fetchForm();
...
+
+// Web Form expects host application to provide a function that it can use to fetch attachments.
+// Signature of the function is (url) => Response; where Response is subset of `fetch` response,
+// It mainly uses `text()` to read the data. We can also pass response `headers`, which can help
+// it to determine the content type but currently that functionality seems to be broken:
+// https://github.com/getodk/web-forms/pull/259#discussion_r1887227207
+const getAttachment = (url) => request({
+  url: apiPaths.formDraftAttachment(props.projectId, props.xmlFormId, false, url.pathname.substring(1))
+}).then(r => ({
+  text: () => (typeof (r.data) === 'string' ? r.data : JSON.stringify(r.data))
```
[the remark this answers is not in the document]
Comment (sadiqkhoja) [reply src/components/form/preview.vue:98]: For geojson, it's an object

**Comment by sadiqkhoja** [comment]
Just noticed that attachments that are linked to dataset are not handled, working on it

File: src/components/form/preview.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthew-white) [reply src/components/form/preview.vue:98]: What will happen with images and other binary attachments? Or are we not worrying about that yet?

What do you think about using the `Content-Type` header to drive some of the logic here? Or is it more so the responsible of Web Forms to call `text()` only if it expects a text response?

```js
text: () => {
  if (typeof (r.data) === 'string') return r.data;
  if (r.headers['content-type'] === 'application/json') return JSON.stringify(r.data);
  throw new Error('response data is not a known text format');
}

[the remark this answers is not in the document]
Comment (sadiqkhoja) [reply src/components/form/preview.vue]: Right now Web Form doesn't support images/binary attachments. Once it's there we would need to pass `Blob` in the `Response` constructor.

File: src/components/form/preview.vue [hunk 2]
```diff
@@ -84,6 +86,53 @@ fetchForm();
...
+  } else {
+    // eslint-disable-next-line no-console
+    console.error('response data is not a known text format');
+  }
+
+  return new Response(body, {
+    status,
+    statusText,
+    headers: fetchHeaders,
+  });
```
Comment (sadiqkhoja) [inline src/components/form/preview.vue:134]: I looked for any option to get [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response) object directly from Axios but couldn't find one. They have recently [implemented](https://github.com/axios/axios/releases/tag/v1.7.0-beta.0) `fetch` adaptor but doesn't provide any way to get original fetch Response

File: src/composables/request.js [hunk 3]
```diff
@@ -111,7 +114,9 @@ const _request = (container, awaitingResponse) => (config) => {
       awaitingResponse.value = false;
 
       logAxiosError(logger, error);
```
Comment (matthew-white) [inline src/composables/request.js:116]: Would you mind including `logAxiosError()` in the `if`? That's what we do in `requestData`.

File: src/composables/request.js [hunk 4]
```diff
@@ -83,13 +83,16 @@ const _request = (container, awaitingResponse) => (config) => {
...
+    alert: alertOption = true,
     ...axiosConfig
   } = config;
 
   const { data } = axiosConfig;
   // This limit is set in the nginx config. The alert also mentions this number.
   if (data != null && data instanceof File && data.size > 100000000) {
-    alert.danger(i18n.t('mixin.request.alert.fileSize', { name: data.name }));
+    if (alertOption) {
+      alert.danger(i18n.t('mixin.request.alert.fileSize', { name: data.name }));
```
Comment (matthew-white) [inline src/composables/request.js:94]: In this case, I think we always want to show the alert. The user will need to take action if their file is too large.

**Review by matthew-white — APPROVED** [review]
Looks great!

**Pull request merged**